rosidl_generator_py/resource/_msg_support.c.em

```diff
@@ -80,130 +81,126 @@ full_classname = '%s.%s._%s.%s' % (spec.base_type.pkg_name, subfolder, module_na
         }
       }
       if (!class_name || !module_name) {
-        return NULL;
+        return false;
       }
       snprintf(full_classname_dest, sizeof(full_classname_dest), "%s.%s", module_name, class_name);
     }
     assert(strncmp(
         "@(full_classname)",
         full_classname_dest, @(len(full_classname))) == 0);
   }
-  @(msg_typename) * ros_message = @(msg_typename)__create();
+  @(msg_typename) * ros_message = _ros_message;
 @[if not spec.fields]@
   (void)ros_message;
 @[end if]@
 @[for field in spec.fields]@
   {  // @(field.name)
     PyObject * field = PyObject_GetAttrString(_pymsg, "@(field.name)");
     if (!field) {
-      return NULL;
+      return false;
     }
 @[  if not field.type.is_primitive_type()]@
 @{
 nested_type = '%s__%s__%s' % (field.type.pkg_name, 'msg', field.type.type)
 }@
-    typedef PyObject *(* convert_from_py_signature)(void *);
+    typedef bool (* convert_from_py_signature)(PyObject *, void *);
     convert_from_py_signature convert_from_py = NULL;
     // get conversion function
     {
       PyObject * msg_module = PyImport_ImportModule("@(field.type.pkg_name).msg._@convert_camel_case_to_lower_case_underscore(field.type.type)");
       if (!msg_module) {
         Py_DECREF(field);
-        return NULL;
+        return false;
       }
       PyObject * msg_class = PyObject_GetAttrString(msg_module, "@(field.type.type)");
       Py_DECREF(msg_module);
       if (!msg_class) {
         Py_DECREF(field);
-        return NULL;
+        return false;
       }
       PyObject * msg_metaclass = PyObject_GetAttrString(msg_class, "__class__");
       Py_DECREF(msg_class);
       if (!msg_metaclass) {
         Py_DECREF(field);
-        return NULL;
+        return false;
       }
       PyObject * convert_from_py_ = PyObject_GetAttrString(msg_metaclass, "_CONVERT_FROM_PY");
       Py_DECREF(msg_metaclass);
       if (!convert_from_py_) {
         Py_DECREF(field);
-        return NULL;
+        return false;
       }
       convert_from_py = (convert_from_py_signature)PyCapsule_GetPointer(convert_from_py_, NULL);
       Py_DECREF(convert_from_py_);
       if (!convert_from_py) {
         Py_DECREF(field);
-        return NULL;
+        return false;
       }
     }
 @[    if field.type.is_array]@
     PyObject * seq_field = PySequence_Fast(field, "expected a sequence in '@(field.name)'");
     if (!seq_field) {
       Py_DECREF(field);
-      return NULL;
+      return false;
     }
 @[      if field.type.array_size is None or field.type.is_upper_bound]@
     Py_ssize_t size = PySequence_Size(field);
     if (-1 == size) {
       Py_DECREF(seq_field);
       Py_DECREF(field);
-      return NULL;
+      return false;
     }
     if (!@(nested_type)__Array__init(&(ros_message->@(field.name)), size)) {
```

Comment:
Still wrapping my head around the message generation code. What checks that `ros_message` is not `NULL` before this point?